zipline/assets/asset_db_migrations.py

```diff
@@ -10,6 +10,53 @@
 from zipline.utils.sqlite_utils import coerce_string_to_eng
 
 
+def alter_columns(op, name, *columns, **kwargs):
+    """Alter columns from a table.
+
+    Parameters
+    ----------
+    name : str
+        The name of the table.
+    *columns
+        The new columns to have.
+    selection_string : str, optional
+        The string to use in the selection. If not provided, it will select all
+        of the new columns from the old table.
+
+    Notes
+    -----
+    The columns are passed explicitly because this should only be used in a
+    downgrade where ``zipline.assets.asset_db_schema`` could change.
+    """
+    selection_string = kwargs.pop('selection_string', None)
+    if kwargs:
+        raise TypeError(
+            'alter_columns received extra arguments: %r' % sorted(kwargs),
+        )
+    if selection_string is None:
+        selection_string = ', '.join(column.name for column in columns)
+
+    tmp_name = '_alter_columns_' + name
+    op.rename_table(name, tmp_name)
+
+    for column in columns:
```

Comment:
Can we add a comment explaining what's happening in this loop? It looks like we're assuming that all columns might have indices on them, and just deleting any of them. I think that means this would break if we used a different index naming scheme from sqlalchemy's default?